src/algebra/jordan/basic.lean

```diff
@@ -0,0 +1,217 @@
+/-
+Copyright (c) 2021 Christopher Hoskin. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Christopher Hoskin
+-/
+import algebra.ring.basic
+import algebra.lie.of_associative
+import data.real.basic
+import linear_algebra.basic
+
+/-!
+# Jordan algebras
+
+We define a set of conditions, formulated in terms of commuting multiplication operators, for the
+multiplication on a non-unital, non-associative semiring to be a Jordan multiplication. When the
+multiplication is commutative, these take a particularly simple form.
+
+A real Jordan algebra `A` can be introduced by
+```
+variables {A : Type*} [non_unital_non_assoc_ring A] [module ℝ A] [smul_comm_class ℝ A A]
+  [is_scalar_tower ℝ A A] [comm_jordan A]
+```
+
+## Main results
+
+- lin_jordan : Linearisation of the commutative Jordan axiom
+
+## Implementation notes
+
+We shall primarily be interested in linear Jordan algebras (i.e. over rings of characteristic not
+two) leaving quadratic algebras to those better versed in that theory.
+
+The conventional way to linearise the Jordan axiom is to equate coefficients (more formally, assume
+that the axiom holds in all field extensions). For simplicity we use brute force algebraic expansion
+and substitution instead.
+
+## References
+
+* [Hanche-Olsen and Størmer, Jordan Operator Algebras][hancheolsenstormer1984]
+* [Cabrera García and Rodríguez Palacios, Non-associative normed algebras. Volume 1]
+  [cabreragarciarodriguezpalacios2014]
+-/
+
+set_option old_structure_cmd true
+
+section non_unital_non_assoc_ring
+
+variables {A : Type*} [non_unital_non_assoc_semiring A]
+
+namespace non_unital_non_assoc_ring
+/--
+Left multiplication operator
+-/
+@[simps] def L : A →+ add_monoid.End A := add_monoid_hom.mul
+
+/--
+Right multiplication operator
+-/
+@[simps] def R : A→+(add_monoid.End A) :=
```

Comment:
This is just `L.flip`

Reply:
Not quite - `@[simps] def R : A→+(add_monoid.End A) := L.flip` gives me
```
type mismatch, term
  L.flip
has type
  A →+ A →+ A : Type (max ? ?)
but is expected to have type
  A →+ add_monoid.End A : Type u_1
```

Reply:
Does this work?
```suggestion
@[simps] def R : A→+(add_monoid.End A) :=
add_monoid_hom.flip (L : A →+ add_monoid.End A)
```

Reply:
Yes, it seems to. Thanks.